rflx/pyrflx/typevalue.py

```diff
@@ -260,3 +460,350 @@ def accepted_type(self) -> type:
     @property
     def literals(self) -> Mapping[Name, Expr]:
         return {}
+
+    def assign(self, value: bytes, length: int = 0, check: bool = True) -> None:
+        """
+        :param value: Bitstring representation of the message
+        :param length: use length parameter if complete
+        length of the message is less than 1 Byte
+        :param check:
+        :return:
+        """
+
+        msg_as_bitstr: Bitstring = Bitstring().from_bytes(value)
+        self.assign_bitvalue(msg_as_bitstr, length)
+
+    def assign_bitvalue(self, value: Bitstring, length: int = 0) -> None:
+        """
+        :param value: Bitstring representation of the message
+        :param length: use length parameter if complete
+        length of the message is less than 1 Byte
+        """
+
+        current_field_name = self._next_field(model.INITIAL.name)
+        field_first_in_bitstr = 0
+        field_length = 0
+
+        while current_field_name != model.FINAL.name and (
+            field_first_in_bitstr + field_length
+        ) <= len(str(value)):
+
+            current_field = self._fields[current_field_name]
+            if isinstance(current_field.typeval, OpaqueValue) and not self._has_length(
+                current_field_name
+            ):
+
+                self._fields[current_field_name].first = self._get_first(current_field_name)
+                self.set(current_field_name, value[field_first_in_bitstr:])
+                self._fields[current_field_name].length = Number(current_field.typeval.length)
```

Comment:
For opaque types there is a different. `Field` contains the length expression derived from the message, e.g. a length field while `typeval` contains the actual length of the buffer. If the field is not set `Length` might be set to the length determined by the message while `typeval.size` is not initialized.

Reply:
Wouldn't it be better to handle this special case in `OpaqueValue` instead of storing `length` in `Field` for all possible `TypeValue`? Could this maybe simplify code on other places?

Reply:
One would have to try if it simplifies code. I'm not hundred percent sure about that.

Reply:
I removed the length from Field, it is now only kept in field.typeval.size. As a result i was able to significantly simplify the set() method of MessageValue but had to add some complexity to ArrayValue and OpaqueValue as the length checking still needs to be done.  
Changes were introduced in e257312